LLama/Native/SafeLLamaSamplerHandle.cs

```diff
@@ -589,5 +741,4 @@ public interface ICustomSampler
     /// Free all unmanaged resources held by this sampler
     /// </summary>
     void Free();
```

Comment:
Maybe rename to `Dispose()` and have `ICustomSampler` implement `IDisposable`?
..and potentially add some automated version of disposing it (?)

I'm aware this is kinda edge case though quite useful still.

Reply:
Good idea, I've made that change with `IDisposable`.

> and potentially add some automated version of disposing it (?)

It's kinda automated at the moment. When the pipeline is disposed all of the samplers are destroyed, including any custom ones implemented in C#. Ultimately it's all owned by `SafeLLamaSamplerChainHandle`, as long as that is disposed (or it's finalizer runs) all of the pipeline resources will be safely destroyed. i.e. the GC is the ultimate fallback thanks to the finalizer.